### PR TITLE
Added differentiation by tuples

### PR DESCRIPTION
**Pull request by Upabjojr**
See discussion: https://groups.google.com/forum/#!topic/sympy/wEgv5w_wCEw

### CODE REVIEW

**Comment by Upabjojr** [comment]
Differentiation by lists does not work as they are not sympified.

**Comment by Upabjojr** [comment]
Probably needs edits to make it compatible with https://github.com/sympy/sympy/issues/5575

File: sympy/core/function.py [hunk 1]
```diff
@@ -1139,13 +1139,16 @@ def __new__(cls, expr, *variables, **assumptions):
         while i < len(variables) - 1:  # process up to final Integer
             v, count = variables[i: i + 2]
             iwas = i
+            if isinstance(v, (collections.Iterable, Tuple)):
```
Comment (asmeurer) [inline sympy/core/function.py]: A less drastic change would be to require Matrix or Array here.

Comment (Upabjojr) [reply sympy/core/function.py]: Matrix and Array are already supported, this is not needed for them.

**Pull request merged**